### ISSUE TITLE
Using prompt.py broke since v0.30.0

### ISSUE THREAD

**Issue by fvaleye** [promptfoo/promptfoo#320]
Hello 👋,

I've encountered an issue starting from version v0.30.0 where the functionality to execute prompts from Python seems to be malfunctioning. Instead of processing the intended output, the model is responding with the raw contents of the prompt's codebase.

My configuration:
```yaml
description: description
providers: openai:gpt-4
prompts: [prompt.py]
tests: tests_prompt.csv
```

My prompt.py file:
```python
def generate_prompt(context) -> str:
    return get_prompt_from(
        location=os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
        file_name="prompt.yaml",
    ) # return a str


if __name__ == "__main__":
    print(generate_prompt(json.loads(sys.argv[1])))
```

However, when I execute the evaluation, I encounter an error. The response from the language model pertains to the prompt.py code itself rather than the expected output from the script.

Thank you!

**Comment by typpo** [comment]
Hi @fvaleye,

Can you try with 0.31.0 and see if that solves the issue?  There were some fixes between then and now.

Here's my standalone test case, which seems to be working fine:

promptfooconfig.yaml:
```yaml
providers: openai:gpt-4
prompts: [prompt.py]
tests: tests.csv
```
prompt.py:
```py
import sys
import json

def generate_prompt(context) -> str:
    return 'write a sentence about ' + context['vars']['topic']

if __name__ == "__main__":
    print(generate_prompt(json.loads(sys.argv[1])))
```
tests.csv:
```
topic
bananas
avocado toast
new york city
```

**Comment by fvaleye** [comment]
Hey @typpo 👋,

It won't work either!

my test.csv file:
```csv
question,context,chat_history,__expected
Who are you?,,,llm-rubric:a chatbot
```

**Comment by typpo** [comment]
Should be fixed with 0.31.1 - please give it a try

**Comment by fvaleye** [comment]
It works now, thanks @typpo 🙏
